staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -37,42 +37,62 @@ import (
 	"k8s.io/klog"
 )
 
-// main demonstrates a leader elected process that will step down if interrupted.
+func buildConfig(kubeconfig string) (*rest.Config, error) {
+	if kubeconfig != "" {
+		cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
 func main() {
 	klog.InitFlags(nil)
+
+	var kubeconfig string
+	var resourceName string
+	var ttlSeconds int
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.StringVar(&resourceName, "resource-name", "example", "The leader election resource name (Configmap or Endpoint name)")
+	flag.IntVar(&ttlSeconds, "ttl", 30, "TTL for leader election in seconds")
```

Comment:
were there issues with the existing 60/15/5 ttl/renew/retry distribution in the example?